add cache control to s3 upload

jschutijzer opened the pull request:
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Add cache control to S3 upload

* **What is the current behavior?** (You can also link to an open issue here)
Cache control on metadata is impossible to set

* **What is the new behavior (if this is a feature change)?**
Added cache control as field in the S3 upload call and passed field to the aws s3 upload sdk

* **Does this PR introduce a breaking change?** (What changes might users need to make in their setup due to this PR?)
No breaking change

jschutijzer commented:
I see i have so different formatting style, so i hope the review process will go ok.

hoegertn commented:
Thanks for the pull request. The formatting makes it hard to review the code. Any chance to fix this?

Can you please add the functionality to the readme and the changelog?

jschutijzer commented:
@hoegertn formatting is better, but can't get it all right without fiddling to much with settings in idea. But I think it's better know. Most things are reordening according to Intellij settings. You can focus on code about cache control. I hope it's ok for you. 
Rebased on master and added comment to changelog to.

hoegertn commented:
Thanks, looks good. I will run a reformat once it is merged and see what happens.
Can you please add the field description to the help pages in the resource folder?

jschutijzer commented:
@hoegertn added field description to help pages. 
Any idea when the next release will be

hoegertn commented:
I try to release today.
